### DISCUSSION TITLE
Trust USB Power QR Code Works for Hero 11 But NOT Hero 12?

### DISCUSSION THREAD

**Discussion by ValinHawK** [gopro/labs#599] [Q&A] [answered]
Hello,

The Trust USB Power (TUSB=1) QR Code works for all my Hero 11 cameras but none of my new Hero 12 cameras.  The 12's respond to scanning the QR Code, "QRCode non-GoPro: TUSB=1".  Any suggestions?

Thanks for any help!

ValinHawK

![image](https://github.com/gopro/labs/assets/143295786/5752a920-8d12-4124-a304-73b7b8c9e3fc)

**Comment by dnewman-gpsw** [comment]
That is not a command, and it will work on any GoPro camera.

No commands start with a capital letter.  

Info on the command language: https://gopro.github.io/labs/control/tech/

**Comment by dnewman-gpsw** [comment]
All Labs commands start with one of this characters:
 - 'a-z', lower case letter for settings -  e.g r4p60 
 - '!' for actions - e.g. !S (or !MTUSB=1 this is what you where looking for.)
 - '>' or '<' for conditions  e.g. >10:00!S
 - '"' for prints. e.g. "Hello World"
 - '=' for assignments, e.g. =Z5

**Reply by ValinHawK** [reply]
Thank you, David.  I'm new to the GoPro world and I see from other posts you are "The Expert" -- thank you in advance for the help!  

I don't know the difference between "Commands" and "Actions" in this context and it seems there are also mode changes one can apply -- maybe that's what Commands do?

I now own five Hero 11's and three 12's (plus an 11 Mini that just failed for the second time on a trip just completed).  The "TUSB=1" QR code is read by the Hero 11s just fine and they boot up when USB power is applied.  I think I originally found this "instruction" on this "Extensions and Other Additions" page, toward the bottom:

https://[gopro.github.io/labs/control/extensions/](https://gopro.github.io/labs/control/extensions/)

Here's the paragraph on it:

_"TUSB=1 - Trust USB power. Some USB power sources may report less than they are capable. This modification assumes the USB Power source is 2A minimum, and disables the testing. This can help with some USB power sources that the camera can reject, but are otherwise sufficient to run all camera operations. If you use TUSB with an inadequate power source, expect capture failures."_

I tested "!MTUSB=1" on one of my 12's and it reads the QR Code fine but still says it does not trust the USB power supplied and shuts down without a battery in it.  With the battery installed, it will not charge the battery while it's recording, only when it is not recording.  If I work around this problem by keeping a battery in the camera, I really need it to charge while recording the way the 11's do, because I need them to record for hours.

To explain, my GoPro camera application is for an airplane in flight, with four of the eight cameras outside.  I designed and installed a power bus just for all the cameras so I would not be distracted trying to manage their battery life while flying the airplane and so they could record long flights up to four hours, well beyond what the batteries can provide.  Of course, I cannot change the batteries in the external cameras in flight...

Here's an example screenshot from one of the cameras (Hero12 with Max Lens 2.0) on my first trip just completed with all these cameras on my airplane:

![image](https://github.com/gopro/labs/assets/143295786/551c2b14-c77d-407d-a645-45ae36112048)

I will see if one of the other 12s will respond as I need to the "!MTUSBB=1" command and trust the USB power supply and report back.  Any other ideas on how to resolve my issue?

Thanks!

**Reply by ValinHawK** [reply]
I tested a second Hero 12 with the !MTUSB=1 command and it does not trust USB power, either.  :(

**Reply by dnewman-gpsw** [reply]
TUSB=1 is not a command on any camera, but if you are so sure, please ignore everything I say.  

If you look at the QR code (extensions page) it will add 'o' and '!' in-front for temporary or permanent. 

HERO12 running current firmware 1.12.70  works with !MTUSB=1 and powers up without a battery.  Check your firmware version.

**Reply by ValinHawK** [reply]
Oh, thanks, I see at the top of the Extensions page that "!M" needs to be put in front of the TUSB=1.  So, I'm surprised it worked without it on the 11's and is not working on my 12's.  I am running the latest Labs firmware 1.12.70.  This is the QR Code I used and gets read okay but is not changing to the Trust USB power mode on my 12s.

![image](https://github.com/gopro/labs/assets/143295786/7443fcb9-3ab2-4472-96bb-1131baa58ab6)

Maybe my 12's have some kind of unique problem.  I have until Wednesday to return them if I can't get the Trust USB power mode to work.  

Thanks again for your help!

**Reply by ValinHawK** [reply]
First level tech support researched the problem and said a patch is needed on the 12's for !MTUSB=1 to work and next level engineering support would be back to me soon with the patch.   Maybe this patch is already on your 12's, Dave...?

**Reply by dnewman-gpsw** [reply]
No the current build is the latest, and trust USB is working correctly. There is no patch, as none is needed.

https://github.com/gopro/labs/assets/10856084/1fa18f04-deee-438d-a052-0fa1ccec4d4c

**Reply by dnewman-gpsw** [reply]
Sorry, the link I sent was from HERO11, HERO12 is fine as you can see in the video above.

**Reply by ValinHawK** [reply]
David, when I got to my computer this morning I was very happy to see your quick reply and the demonstration video of the Trust USB Power code working on a Hero 12 -- thank you!  So, of course I realized I should make a video showing the problem I'm having with my three new Hero 12s trusting the USB power supplied.  Here's a link to my demo video:

[Valin GoPro Hero 12 Trust USB Power Problem Demo](https://photos.app.goo.gl/y54pD4fgiuhT4mK1A)

I show in the video how the 12 has the latest 01.12.70 firmware, that it accepts the !MTUSB=1 QR Code, and does not trust the USB power on boot up.  Next, I show the 11 accepts the USB power supply.   Then, with everything the same, except using a GoPro branded USB cable, the 12 powers up and accepts the USB power!?

I don't know enough about what info is extracted by devices on both sides of a USB cable connection to understand why the cable makes a difference for the 12s but not the Hero 11s.  Of course, there's evidently something in the firmware that's getting tripped up by what it's reading about the cable or its specifications.

The USB cables we use in our airplane's new GoPro camera system are of high quality (better than the generic one used in my demo) and work with the Hero 11's just fine.

I hope this info provides some helpful clues to help resolve this problem?   Your thoughts...?

Thanks!

Valin

![image](https://github.com/gopro/labs/assets/143295786/41190ba9-47f8-463b-a300-8b7765d8e2a5)

**Reply by dnewman-gpsw** [reply]
The video is showing that you don't have TUSB enabled.  It should be listed on the "Features Activated:" screen during boot. 
![image](https://github.com/gopro/labs/assets/10856084/07a1b113-db6f-4a16-a30d-fe08e845df11)

This is likely a setting metadata corruption (maybe the long OWNR info caused a problem.)  Scan !RESET (press the shutter 3 times quickly) then scan your settings again, PRXY, TUSB, etc.  Add the OWNR last.  This shouldn't normally be a requirement, but I'm guessing based on your features screen.

The reason the GoPro cable is working, is that Apple power source with a full spec cable, is communicating the supply abilities. TUSB is for bad or long cables, or older power sources that don't communicate their ratings.

**Reply by ValinHawK** [reply]
Thanks Dave.  I tried resetting the firmware and rescanning my codes, completely leaving out the Owner Info QR Code.  I think, though, that the !RESET did not work because it did not beep when being scanned and I noticed the Trust USB Power (!MTSUB=1) did not beep on scanning, either.  

Here's the video we made doing all that in case it's helpful.  I'm pretty sure the !RESET QR Code did not work because when I pressed the shutter button three times quickly, after scanning it, all it seemed to do was start a recording and stop it...  I'll await your additional suggestions.  Thanks!

https://photos.app.goo.gl/Bs9hVrGX5BoHiYQJA

**Reply by dnewman-gpsw** [reply]
You didn't successfully reset it.  

https://github.com/gopro/labs/assets/10856084/ca670033-34aa-473b-aac3-174d901cb265

If you don't see "approved", you didn't press three buttons in time.
And after a power cycle you should see "GoPro Labs Beta Firmware, Participation Terms Apply", and with no Feature Activated.

**Reply by ValinHawK** [reply]
Ours never showed "QR Data Reset?" after scanning the reset code.  I notice your QR Code is different than mine.  Is there a different QR Code generator I need to use?

![image](https://github.com/gopro/labs/assets/143295786/6ee0ca51-3919-4bdc-b722-e01753759c5d)

**Reply by ValinHawK** [reply]
I used the QR Code generator on the Lab's Settings page:

![image](https://github.com/gopro/labs/assets/143295786/311cea3d-c986-420b-9acd-1455f1e42be9)

**Comment by dnewman-gpsw** [comment]
There must be a keyboard extension or language profile that I can't see that is adding invisible/invalid characters. This could be way your camera are behaving weirdly. Try using the QRControl app, it has !RESET in the extras menu, and it removed you computer from the setup.

When I type !RESET I get this:
![image](https://github.com/gopro/labs/assets/10856084/cd6b6daf-6c0f-4012-b4b8-c416f4536285)

Scan this image.

**Reply by dnewman-gpsw** [reply]
You !RESET has a space or another non-display character before the !.  QR codes must start with ! or a lower case character.

**Reply by dnewman-gpsw** [reply]
Use copy as URL https://gopro.github.io/labs/control/set/?cmd=!RESET  as it will show all the character(s) you are adding.

**Reply by ValinHawK** [reply]
Oh, my bad for trying to copy and paste from our chat.  I'll try again.

**Reply by ValinHawK** [reply]
Well, I found out why the !MTUSB=1 wasn't working/beeping on scan -- because of the same problem, there was a blank space in front of it for the QR Code generation.  The QR Code was corrected and scanned and it beeped and shows TUSB is an active feature on the boot.  

Now a new problem has revealed itself... :(

The Hero 12 camera boots up without a battery with only USB power just fine. But, when I press record it starts recording and then the screen goes black.  When I power it back up it shows it's repairing the video file and succeeds.  Progress in that it boots but no recording capability...

So, I wondered about my Hero 11 cameras which I now realize never had trust USB power feature working -- but was somehow doing fine on USB power alone.  So, I took one of the 11's and scanned the !MTUSB=1 QR Code and it beeped and accepted it.  I did the same test, USB power, no battery...  It booted up fine, showed on the boot screen TUSB.  When I pressed record, now it crashed to power off the same way as the Hero 12...!

We're making progress...  Any suggestions for what's causing this problem?

Thanks!

**Reply by ValinHawK** [reply]
I should emphasize that the same USB cable was working and recording on the Hero 11 with no battery installed before I scanned the !MTUSB=1 QR Code...

**Reply by dnewman-gpsw** [reply]
I think you should just experiment more, there is a lot of experience to gain. Try different modes, and settings. You saw it working on my camera, there haven't been on other reports of this. I can't guess what you are doing differently, as so you have so far surprised me with user errors I hadn't expected. Labs is not user friendly, but it would have much less if is was, so that is the trade off.

**Reply by ValinHawK** [reply]
Now I've torpedoed one of my Hero 11's that was working fine on USB power.  I don't know what to do.  My window on returning the 12's closes tomorrow.  So, I think I'm going to have to return them tomorrow if I can't figure out how to resolve this current problem. :(  On my 11, I'll try resetting it and NOT doing the Trust USB feature.

When you tested your 12 did you make sure it was using a USB cable that would normally not allow the camera to work without the Trust USB Power feature active...?

Thanks!

**Comment by dnewman-gpsw** [comment]
There is no technical reason to return the cameras.

Here is a Trust USB demo before and after scanning.  

https://github.com/gopro/labs/assets/10856084/fc06e465-25df-4bb4-afe6-05ea0a30fd55

**Reply by ValinHawK** [reply]
On my poor USB cable it is shutting down after pressing record.  And my Hero 11 was recording fine on that cable alone until I scanned the Trust USB Power QR Code and now it shuts down on record, too.  Should I send a video showing it on another of my Hero 11's?

I think tomorrow I'll put the cameras back on the airplane and test how they do on its camera power system and cables before returning the three new Hero 12s.

Thank you for your help.  I would never have figured out two of my QR Codes were messed up with blank characters at the front if I hadn't seen your video where your QR Code pattern was different than mine and you explained why...

**Reply by ValinHawK** [reply]
Dave, I just swapped out my crappy cable with different crappy USB cable, and the cameras are recording fine with it.  So, my first crappy USB cable must have a problem (maybe bad high resistance connection) with it and I've now thrown it away.  

So, good news is you have really helped this Noob get over the hump on using these special commands/actions/modes on your Labs Beta Firmware which I really need for recordings on our flights.  I'll still go confirm they all work fine on the airplane's power system tomorrow.  

Thanks again for your expert help!

Valin

**Reply by dnewman-gpsw** [reply]
Trust USB does not anything to effect camera behavior, 11 or 12.  All is does is report a maximum current, if the power supply doesn't. If the cable is really bad, voltage can drop over the cable, capture can fail, nothing to do with TrustUSB.  With TUSB, **you** are certifying that your power and cables are good.  

This is all the code involved:
`if(g_trust_usb_power) *max_current = 3000; //mA`

**Reply by dnewman-gpsw** [reply]
Yep, crappy cable will mean crappy results.

**Reply by ValinHawK** [reply]
Yes!  That first crappy cable was working fine originally but I expect from all my flexing and fiddling with it that it started to lose its electrical integrity and was causing too much of a power drop when the cameras tried to record and needed more power.  It was just unfortunate coincidence that the cable degraded enough that it failed in that way just as the QR Code issue was worked out.  Thanks again!

**Reply by ValinHawK** [reply]
Hey Dave, we’re at the airplane with all cameras installed. All the cameras now trust USB power. The Hero 12’s, though, are not booting up on USB power trigger.  The 11’s wake up and connect to the Quick app. 

Did I create this QR Code correctly?  I dont have an action I want the camera to perform after booting up. We just need the cameras to boot up on there own without a manual push of each camera’s power button and go under Quick app remote control. 

Here’s a screenshot of the QR Code. 
![IMG_6532](https://github.com/gopro/labs/assets/143295786/f1a481f8-3454-470e-83f4-794bea5c5e7e)

We will wait at our hangar hoping you see this soon… 🤪

Thanks!

PS:  It’s okay to call me at 281-486-4663 if that’s easiest.

**Reply by dnewman-gpsw** [reply, marked as answer]
I wouldn't use the old USB Power Trigger, as it is not reliable, and it was designed to require a battery. Setting !MWAKE=2 is way better. With this command the camera will always boot when power is added. You can also combine with boot commands so the camera will do whatever you want on start up.

**Reply by ValinHawK** [reply]
Thanks!  We’ll try that command here in minutes.

**Reply by ValinHawK** [reply]
Thanks Dave!  That is the perfect mode command for bringing up all the cameras remotely and them automatically connecting to the Quick app.  

All these things you’ve helped with are going to make it so so much easier to manage the cameras on our flights.  What a relief for us as the difficulties we’ve had on our camera test flights and first long trip were very frustrating and dangerously distracting to piloting the airplane. 

Thanks again!!!
